hazelcast-windows-vm-cluster/azuredeploy.json

```diff
@@ -0,0 +1,292 @@
+          }
+      },
+      "adminPassword": {
+          "type": "SecureString",
+          "metadata": {
+              "description": "Password for hazelcast VMs"
+          }
+      },
+      "hazelcastVersion": {
+        "defaultValue": "3.11",
+        "type": "String",
+        "metadata": {
+            "description": "Hazelcast Version"
+        },
+        "allowedValues": [
+            "3.11",
+            "3.11.1"
+          ]
+    },
+      "clusterName": {
+          "defaultValue": "[parameters('adminUsername')]",
+          "type": "String",
+          "metadata": {
+              "description": "Username for hazelcast grid"
+          }
+      },
+      "dnsPrefix": {
+          "type": "String",
+          "metadata": {
+              "description": "Unique DNS Name for the Public IP used to access the Virtual Machine."
+          }
+      },
+      "instanceCount": {
+          "type": "Int",
+          "metadata": {
+              "description": "The number of hazelcast nodes in the grid"
+          }
+      },
+      "clusterTag": {
+          "type": "String",
+          "metadata": {
+              "description": "The tag used to identify this cluster"
+          }
+      },
+      "clusterPort": {
+          "defaultValue": "5701",
+          "type": "String",
+          "metadata": {
+              "description": "The cluster port to identify this cluster"
+          }
+      },
+      "aadClientId": {
+          "type": "String",
+          "metadata": {
+              "description": "Azure Active Directory Service Principal client ID"
+          }
+      },
+      "aadClientSecret": {
+          "type": "String",
+          "metadata": {
+              "description": "Azure Active Directory Service Principal client secret"
+          }
+      },
+      "aadTenantId": {
+          "type": "String",
+          "metadata": {
+              "description": "Azure Active Directory Service Principal tenant ID"
+          }
+      },
+      "vmSize": {
+          "defaultValue": "Standard_DS2",
+          "type": "String",
+          "metadata": {
+              "description": "The size of each instance VM"
+      },
+          "allowedValues": [
+            "Standard_DS1",
+            "Standard_DS2",
+            "Standard_DS3",
+            "Standard_DS4",
+            "Standard_DS8",
+            "Standard_DS16",
+            "Standard_D1"
+          ]
+      },
+      "windowsOSVersion": {
+          "defaultValue": "2016-Datacenter",
+          "type": "String",
+          "metadata": {
+              "description": "The Windows version for the VM. This will pick a fully patched image of this given Windows version."
+          },
+          "allowedValues": [
+            "2008-R2-SP1",
+            "2012-Datacenter",
+            "2012-R2-Datacenter",
+            "2016-Nano-Server",
+            "2016-Datacenter-with-Containers",
+            "2016-Datacenter"
+          ]
+      },
+      "templateBaseUrl": {
```

Comment:
See [BP guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#deployment-artifacts-nested-templates-scripts) for how to handle artifacts